### ISSUE TITLE
Please switch to semantic version tags

### ISSUE THREAD

**Issue by alerque** [knadh/listmonk#343]
The current version tags include the label "`-beta`" as part of the version tag itself. This is common for projects where the release cycle includes actual beta builds of a version and then that same version itself, but that's now what is being done here. This project doesn't release `v0.9.0-beta` and then `v0.9.0`, it just uses the beta label as part of a title and increments the semantic part of the tag numbering. There was never a `v0.8.0`, only a `v0.8.0-beta` then `v0.9.0-beta`.

This makes distro packaging an absolute fiasco, and it even trips up the app's own internal handling of versioning. Right now if I build from Git HEAD version it claims there is a newer version available. If I compile the tagged version with the beta string dropped from the version string myself, it also identifies the upstream beta tag as newer. Both cases are wrong.

Please just stop adding `-beta` to the actual Git tags and use the parse-able semantic `vX.Y.Z` format. Feel free to add "Beta" to the release **titles** on Github, just not the actual git tag. It can also be embedded as part of the BUILDSTR for now, and the Github releases can be left in "pre-release" mode to indicate they are not expected to be stable releases, but please make it easier on actual programming by not making the version tags so hard to parse.

**Comment by knadh** [comment]
hm, suffixes like `-beta` are perfectly [valid semver](https://semver.org/) tags. They're parsed [just fine](https://github.com/knadh/listmonk/blob/68b80d0eb603d6e747284a29540cb0792911cad9/cmd/upgrade.go#L116) by `golang.org/x/mod/semver` as well. That said, with `v1.0.0`, the beta suffix will be dropped (because the beta is ending and not because there's anything wrong with the suffix).

> Right now if I build from Git HEAD version it claims there is a newer version available.
This is because if you're just doing `go build`, the git version is not embedded in the binary. The binary needs to know its version to figure this out. Please check the build flags in the [Makefile](https://github.com/knadh/listmonk/blob/68b80d0eb603d6e747284a29540cb0792911cad9/Makefile#L23).

**Comment by alerque** [comment]
Suffixes like `-beta` are valid semver, but you aren't using them with the sequence that semver would specify!

The internal version check is registering `v0.9.0-beta` as newer than `v0.9.0` (which is backwards per the semver docs you linked), and you are not releasing `v0.9.0` which should come after `v0.9.0-beta`. You are skipping from beta tag to beta tag tag without ever having stable tags for the version numbers that had beta releases.

The current release history suggests you are treating `v0.8.0-beta` as if it was `v1.0.0-beta8`, `v0.9.0-beta` as `v1.0.0-beta9` etc. Personally I dislike the suffixes entirely, but if you did use them at least match them with actual releases of the thing that was previously a beta. I prefer the incrementing numbers anyway, but ditch the suffixes if you aren't actually making pre-releases.
